src/components/FullscreenGrid/styles.module.scss

```diff
@@ -1,5 +1,8 @@
 @import '../../scss/variables';
 
+$collapse-button-height: 40px;
```

Comment:
The idea with the patch is to always render the collapsed state (the tall button panel) within ContentShell but allow the child content to fully control the layout of the expanded state.

Reply:
Oh, hey, you can actually match the button styles just by letting the flex layout calculate everything. Here is a patch with the collapse button matching the accordion buttons exactly:

```diff
diff --git a/src/components/FullscreenGrid/styles.module.scss b/src/components/FullscreenGrid/styles.module.scss
index 7caaca4..3eb4149 100644
--- a/src/components/FullscreenGrid/styles.module.scss
+++ b/src/components/FullscreenGrid/styles.module.scss
@@ -32,7 +32,9 @@ $expand-button-width: 40px;
   margin-left: $default-padding;
 
   .mainSidePanelToggleButton {
-    height: $collapse-button-height;
+    height: auto;
+    margin-top: $default-padding;
+    padding: $default-padding;
     width: 100%;
   }
 
@@ -66,8 +68,13 @@ $expand-button-width: 40px;
   padding: $default-padding;
 }
 
+.mainSidePanel {
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+}
+
 .mainSidePanelContent {
-  height: calc(100% - #{$collapse-button-height + $default-padding});
   overflow-x: auto;
 
   .mainSidePanelIsCollapsed & {
```

(Apparently this problem haunted me over the weekend, ha.)

Reply:
I filed https://github.com/mozilla/addons-code-manager/issues/737 because I could not figure out how to get the same style for all buttons. Thanks for the patch!

To answer the very first question, I'd like to keep the in the side panel and not in the accordion so that it is a fullscreen grid feature and it works out of the box, no matter what we put in the side panels.

Reply:
> To answer the very first question, I'd like to keep the in the side panel and not in the accordion so that it is a fullscreen grid feature and it works out of the box, no matter what we put in the side panels.

Yes, I agree that this is nice. The second patch I gave you allows the buttons to look the same and also keeps the feature encapsulated in `ContentShell`. Did that work for the other screens, too?

Reply:
> Did that work for the other screens, too?

Looks like it works super well!